galsim/wfirst/wfirst_detectors.py

```diff
@@ -82,22 +82,54 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
     """
     img.applyIPC(galsim.wfirst.ipc_kernel, edge_treatment=edge_treatment, fill_value=fill_value)
 
-def allDetectorEffects(img, rng=None, exptime=None):
+def applyPersistence(img, prev_exposures):
+    """
+    Applies the persistence effect to the Image instance by adding a small fraction of the previous
+    exposures (up to 8) supplied as the 'prev_exposures' argument.
+
+    For more information about persistence, see the docstring for galsim.Image.applyPersistence.
+    Unlike that routine, this one does not need the coefficients to be specified. However, the list
+    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
+    ignored.
+
+    @param img               The Image to be transformed.
+    @param prev_exposures    List of up to 8 Image instances in the order of exposures, with the
```

Comment:
Done.

On Tue, Apr 26, 2016 at 5:23 AM, Mike Jarvis <notifications@github.com>
wrote:

> In galsim/wfirst/wfirst_detectors.py
> <https://github.com/GalSim-developers/GalSim/pull/710#discussion_r61053962>
> :
>
> > @@ -82,22 +82,54 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
> >      """
> >      img.applyIPC(galsim.wfirst.ipc_kernel, edge_treatment=edge_treatment, fill_value=fill_value)
> >
> > -def allDetectorEffects(img, rng=None, exptime=None):
> > +def applyPersistence(img, prev_exposures):
> > +    """
> > +    Applies the persistence effect to the Image instance by adding a small fraction of the previous
> > +    exposures (up to 8) supplied as the 'prev_exposures' argument.
> > +
> > +    For more information about persistence, see the docstring for galsim.Image.applyPersistence.
> > +    Unlike that routine, this one does not need the coefficients to be specified. However, the list
> > +    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
> > +    ignored.
> > +
> > +    @param img               The Image to be transformed.
> > +    @param prev_exposures    List of up to 8 Image instances in the order of exposures, with the
>
> Could we make the 8 here and elsewhere a parameter? E.g. put {ncoeff}
> here and then at the end of the string write
> ...""".format(ncoeff=len(galsim.wfirst.persistence_coefficients))
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/710/files/ab9d9f74f9bb2d3de9a54059daec1ac56e23cd7c#r61053962>
>



-- 
Regards

Arun Kannawadi Jayaraman
PhD candidate, Dept. of Physics,
Carnegie Mellon University,
Pittsburgh, PA 15213